Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -136,6 +136,33 @@ public void SerializesAndDeserializesCustomPropertyNames()
             Check(TestObjects.fmRadio);
         }
 
+        [TestMethod]
+        public void SerializesAndDeserializesGetterAndSetterMethods()
+        {
+            Check(TestObjects.SchoolDesk);
+        }
+
+        [TestMethod]
+        public void DoesNotDoubleSerializeIonFieldsAlreadySerializedByMethods()
+        {
+            IIonStruct serialized = ToIonValue(new IonSerializer(), TestObjects.Ruler);
+
+            Assert.IsTrue(serialized.ContainsField("length"));
+            Assert.IsTrue(serialized.ContainsField("unit"));
+
+            // We should have exactly two fields. ie. we did not double serialize the Ruler's length or unit members.
+            Assert.AreEqual(2, serialized.Count);
+        }
+
+        [TestMethod]
+        public void ExceptionOnMultiParameterIonPropertySetterMethods()
+        {
+            var serializer = new IonSerializer();
+
+            var stream = serializer.Serialize(TestObjects.Chalkboard);
+            Assert.ThrowsException<NotSupportedException>(() => serializer.Deserialize<Chalkboard>(stream));
```

Comment:
This isn't a `NotSupportedException` it's rather an `InvalidOperationException`.

Reply:
Addressed.